### PR TITLE
initialize perses CLI

### PR DESCRIPTION
**Pull request by Nexucis**
This PR initialize the Perses CLI with CI, first folder organization, first command: `version`

```bash
./bin/p3s version --help                                       
Display client version

Usage:
  p3s version [flags]

Flags:
  -h, --help            help for version
  -o, --output string   One of 'yaml' or 'json'. Default is 'yaml'
      --short           If true, print just the version number.
```

```bash
./bin/p3s version -oyaml
buildTime: "2022-01-27"
version: ""
commit: 70d5a66a04639bc1d322a95a104623c2dab8bae3
```

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -12,11 +13,32 @@ builds:
       - PKG_LDFLAGS=github.com/prometheus/common/version
     ldflags:
       - "-s -w -X {{.Env.PKG_LDFLAGS}}.Version={{.Version}} -X {{.Env.PKG_LDFLAGS}}.Revision={{.Commit}} -X {{.Env.PKG_LDFLAGS}}.BuildDate={{.Date}} -X {{.Env.PKG_LDFLAGS}}.Branch={{.Branch}}"
+  - id: "p3s"
```
[the remark this answers is not in the document]
Comment (Nexucis) [reply .goreleaser.yml:16]: oh it's just an idea that come from the #perses-dev.

image

File: internal/api/shared/database/dao.go [hunk 2]
```diff
@@ -113,10 +113,10 @@ func (d *fileDAO) Query(query etcd.Query, slice interface{}) error {
 	}
 
 	// it's a pointer, so move to the actual element behind the pointer.
-	// Having a pointer avoid to get the error:
+	// Having a pointer avoid getting the error:
```
[the remark this answers is not in the document]
Comment (Nexucis) [reply internal/api/shared/database/dao.go:116]: it's just a typo. But yeah if you want I can revert them it's not related to this PR in particularly

File: internal/cli/utils/output.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+)
+
+const (
+	JSONOutput = "json"
+	YAMLOutput = "yaml"
+)
+
+func ValidateOutput(o string) error {
+	if o != YAMLOutput && o != JSONOutput {
+		return fmt.Errorf("--ouput must be %q or %q", YAMLOutput, YAMLOutput)
```
[the remark this answers is not in the document]
Comment (Nexucis) [reply internal/cli/utils/output.go:32]: lol, I don't know what I was thinking there. Thanks for your eyes !

File: .goreleaser.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sjcobb) [reply .goreleaser.yml:16]: I don't have a strong opinion, but `percli` isn't bad

[the remark this answers is not in the document]
Comment (sjcobb) [reply .goreleaser.yml:16]: Well not sure it matters, but of course someone [already thought of percli ](https://github.com/headwirecom/percli) :laughing:

[the remark this answers is not in the document]
Comment (Nexucis) [reply .goreleaser.yml:16]: Other proposition could be just `pcli`.
I think for a CLI it's nicer if it's a short name.

But I don't have a strong opinion either

[the remark this answers is not in the document]
Comment (juliusv) [reply .goreleaser.yml:16]: I'd say just go with `percli` for now, doesn't matter if another project with two stars also uses it :P

[the remark this answers is not in the document]
Comment (Nexucis) [reply .goreleaser.yml:16]: alright I will change the name for `percli`.
If we feel it's not a good a name, we can still change it.

**Comment by juliusv** [comment]
:+1: Ship it :ship:

**Pull request merged**